param/__init__.py

```diff
@@ -132,12 +133,20 @@ def named_objs(objlist, namesdict=None):
     """
     objs = OrderedDict()
 
+    objtoname = {}
+    unhashables = []
     if namesdict is not None:
-        objtoname = {hashable(v): k for k, v in namesdict.items()}
+        for k, v in namesdict.items():
+            try:
+                objtoname[hashable(v)] = k
+            except TypeError:
+                unhashables.append((k, v))
 
     for obj in objlist:
-        if namesdict is not None and hashable(obj) in objtoname:
+        if objtoname and hashable(obj) in objtoname:
```

Comment:
```suggestion
        if hashable(obj) in objtoname:
```

(checking that the dict is non-empty doesn't seem like it achieves anything)

Reply:
Sure, it avoids the potentially expensive `hashable` conversion.

Reply:
Ok